### PR TITLE
Cleanup and performance

### PR DESCRIPTION
**Pull request by eriksoe**
Improvements in vclock.

### CODE REVIEW

**Comment by krestenkrab** [comment]
Looks clean except indentation in the prune_vclock fix

File: src/vclock.erl [hunk 1]
```diff
@@ -136,25 +136,25 @@ increment(Node, IncTs, VClock) ->
...
+       ).
+
 % @doc Return a timestamp for a vector clock
 -spec timestamp() -> timestamp().
 timestamp() ->
-    calendar:datetime_to_gregorian_seconds(erlang:universaltime()).
+    %% Same as calendar:datetime_to_gregorian_seconds(erlang:universaltime()),
+    %% but significantly faster.
+    %% Assumes that less than a million now() calls are done per second.
+    {MegaSeconds, Seconds, _} = now(),
```
Comment (Vagabond) [inline src/vclock.erl]: Wouldn't os:timestamp be even faster, it doesn't have to be locked to make the clock monotonic like now().

Comment (eriksoe) [inline src/vclock.erl]: @Vagabond: Ah - didn't know about os:timestamp(), and wondered why it (a fast timestamp function without monotonicity guarantee and the possibility of skewedness it entails) wasn't there.
Thank you!

Comment (eriksoe) [inline src/vclock.erl]: Fwiw, my measurements show that it may be a bit (20-30%) faster - measurable but not dramatical.

**Comment by Vagabond** [comment]
Merged, with some cleanups.

**Pull request closed without merging**